lib/src/content/wolt_modal_sheet_animated_switcher.dart

```diff
@@ -231,6 +231,10 @@ class _WoltModalSheetAnimatedSwitcherState extends State<WoltModalSheetAnimatedS
     final topBarTitle = WoltModalSheetTopBarTitle(page: _page, pageTitleKey: _pageTitleKey);
     final navBarHeight =
         _page.navBarHeight ?? themeData?.navBarHeight ?? defaultThemeData.navBarHeight;
+
+    assert(_page.topBarWidget == null || isTopBarLayerAlwaysVisible,
```

Comment:
If we are introducing a top bar, which is used only when isTopBarLayerAlwaysVisible, should we remove `isTopBarLayerAlwaysVisible` boolean, and instead introduce `customTopBar` which is always visible?

Reply:
I'm probably not the best person to answer this question. Adding a custom top bar when `isTopBarLayerAlwaysVisible` is set to `false` requires many more changes and probably won't play well with all the animations (or, well, it would require the custom top bar to be very carefully designed to fit with the rest of the design).

Reply:
Btw I was a bit undecided should the new parameter be called `topBarWidget` or `customTopBarWidget`. What would you prefer?

Reply:
I think `topBarWidget` is ok.

Reply:
It might be good to add `topBarWidget` parameter to the `withSingleChild` and `withCustomSliverList` constructors, also.

Reply:
@yunusemrebakir There are no such constructors in wolt_modal_sheet_animated_switcher.dart and the parameter have been added in all constructors in wolt_modal_screen_page.dart already :)

Reply:
ah sorry missed it, it is already added indeed 👍🏽